src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,917 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+
+    # Validate the helm environment file for Dogfood.
+    dp_endpoint_dogfood = None
+    release_train_dogfood = None
+    if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.Dogfood_RMEndpoint:
+        dp_endpoint_dogfood, release_train_dogfood = validate_env_file_dogfood(values_file, values_file_provided)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    load_kube_config(kube_config, kube_context)
+    configuration = kube_client.Configuration()
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection(configuration)
+
+    # Get kubernetes cluster info for telemetry
+    kubernetes_version = get_server_version(configuration)
+
+    # Checking helm installation
+    check_helm_install(kube_config, kube_context)
+
+    # Check helm version
+    helm_version = check_helm_version(kube_config, kube_context)
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.HelmVersion': helm_version})
+
+    # Check for faulty pre-release helm versions
+    if "3.3.0-rc" in helm_version:
+        telemetry.set_user_fault()
+        raise CLIError("The current helm version is not supported for azure-arc onboarding. Please upgrade helm to a stable version and try again.")
+
+    release_namespace = validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context)
+
+    # Fetch Connected Cluster for agent version
+    connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
+
+    if hasattr(connected_cluster, 'distribution') and (connected_cluster.distribution is not None):
+        kubernetes_distro = connected_cluster.distribution
+    else:
+        kubernetes_distro = get_kubernetes_distro(configuration)
+
+    if hasattr(connected_cluster, 'infrastructure') and (connected_cluster.infrastructure is not None):
+        kubernetes_infra = connected_cluster.infrastructure
+    else:
+        kubernetes_infra = get_kubernetes_infra(configuration)
+
+    kubernetes_properties = {
+        'Context.Default.AzureCLI.KubernetesVersion': kubernetes_version,
+        'Context.Default.AzureCLI.KubernetesDistro': kubernetes_distro,
+        'Context.Default.AzureCLI.KubernetesInfra': kubernetes_infra
+    }
+    telemetry.add_extension_event('connectedk8s', kubernetes_properties)
+
+    if disable_cluster_connect:
+        try:
+            helm_values = get_all_helm_values(release_namespace, kube_config, kube_context)
+            if not disable_cl and helm_values.get('systemDefaultValues').get('customLocations').get('enabled') is True and helm_values.get('systemDefaultValues').get('customLocations').get('oid') != "":
+                raise Exception("Disabling 'cluster-connect' feature is not allowed when 'custom-locations' feature is enabled.")
```

Comment:
Log telemetry exception & mark user fault here

Reply:
This is basically just a user input validation. So, didnt add telemetry.

And going forward, we have to remove all telemetry as WI. CLI team will add telemetry itself based on type of error. They had asked us to do this GA time. But, I had said we will raise separate pr later.

Reply:
But what about custom telemetry which is specific to our service? Will that also be removed? That dosen't seem right

Reply:
They have created a list of error types 

Say UserAuthError("..") -> raise telemetry too

Say InputError("...") -> raise user fault and normal telemetry too

I have added what the recommendation/requirement is from them, in the haiku one note under 'post GA work'

Reply:
https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md

Reply:
Custom telemetry can be done using customerrortypes, which they have explained in the doc above.

Reply:
Okay then let's capture that in next PR soon

Reply:
Service related telemetry wont have to removed. Only the ones related to CLIError which we do now.